db/migrate/20150331152908_create_neighborhoods.rb

```diff
@@ -0,0 +1,14 @@
+class CreateNeighborhoods < ActiveRecord::Migration
+  def change
+    create_table :neighborhoods do |t|
+      t.string :name, null: false, length: { minimum: 4 }
+      t.string :location, null: false
+      t.text :description
+      t.string :image_url
+      t.integer :user_id, null: false
+      t.index :name, unique: true
```

Comment:
Test to see if this works. If not, use the traditional syntax for creating an index with uniqueness contsraint. Run a new migration with the following:

```ruby
def change
  #drop table syntax here
  add_index :neighborhoods, :name, unique: true
end
```